### PR TITLE
feat: add saml custom attribute action and translations

### LINKED ISSUE #5014: Custom attributes in SAML response
As an administrator, I want to be able to define custom attributes in the SAML response so that I can connect more service providers to ZITADEL, which for example needs the roles of the user.
(called custom claims in OIDC)

**Acceptance criteria**
- [x] SAML responses can be extended with additional attributes
- [x] SAML responses can be modified/extended through actions
- [x] SAML responses can be modified/extended with attributes of different types and names
- [ ] Add Documentation how to do it

### PR DESCRIPTION
**Pull request by stebenz**
### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

closes #5014

### CODE REVIEW

File: internal/api/saml/storage.go [hunk 1]
```diff
@@ -191,3 +209,139 @@ func setUserinfo(user *query.User, userinfo models.AttributeSetter, attributes [
...
+		),
+	)
+
+	for _, action := range queriedActions {
+		actionCtx, cancel := context.WithTimeout(ctx, action.Timeout())
+
+		apiFields := actions.WithAPIFields(
+			actions.SetFields("v1",
+				actions.SetFields("attributes",
+					actions.SetFields("setCustomAttribute", func(name string, nameFormat string, attributeValue []string) {
```
Comment (livio-a) [inline internal/api/saml/storage.go:261]: please change the `attributeValue` to a variadic param

**Review by livio-a — CHANGES_REQUESTED** [review]

**Comment by livio-a** [comment]
BTW i successfully tested the current state with OIDC (access and id_token customization) and SAML (attributes)

File: internal/api/saml/storage.go (hunk 1 above)
Comment (stebenz) [reply internal/api/saml/storage.go]: done

File: docs/docs/apis/actions/customize-samlresponse.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    - `getUser()` [*User*](./objects#user)
+    - `user`
+      - `getMetadata()` [*metadataResult*](./objects#metadata-result)
+      - `grants` [*UserGrantList*](./objects#user-grant-list)
+- `api`  
+  The second parameter contains the following fields:
+  - `v1`
+    - `attributes`
+      - `setCustomAttribute(string, string, Array of string)`  
+        Sets any value as attribute in addition to the default attributes, if the key is not already present. The parameters represent the key, nameFormat and the attributeValue.
```
Comment (livio-a) [inline docs/docs/apis/actions/customize-samlresponse.md:25]:
```suggestion
      - `setCustomAttribute(string, string, ...string)`  
        Sets any value as attribute in addition to the default attributes, if the key is not already present. The parameters represent the key, nameFormat and the attributeValue(s).
```

File: docs/docs/apis/actions/customize-samlresponse.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+title: Customize SAMLResponse Flow
```
Comment (livio-a) [inline docs/docs/apis/actions/customize-samlresponse.md:2]:
```suggestion
title: Complement SAMLResponse
```

**Review by livio-a — APPROVED** [review]

**Pull request merged**